### ISSUE TITLE
Deploying the same ingress gateway in two different namespaces seems to conflict

### ISSUE THREAD

**Issue by yann-soubeyrand** [istio/istio#24638]
**Bug description**

When deploying the same ingress gateway (same deployment and same Gateway manifests, with the exception of the namespace) in two different namespaces, the gateways refuse to start with a message in the logs indicating a configuration conflict.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

The two gateways should be totally independent of each other and there should be no configuration conflict between them.

**Steps to reproduce the bug**

Launch a new Minikube cluster, extract the attached [manifests.zip](https://github.com/istio/istio/files/4771799/manifests.zip) archive and execute:
```
istioctl install -f istio.yaml
kubectl create ns istio-ingressgateway-1
kubectl create ns istio-ingressgateway-2
istioctl manifest generate -f istio-ingressgateway-1.yaml | kubectl -n istio-ingressgateway-1 apply -f -
istioctl manifest generate -f istio-ingressgateway-2.yaml | kubectl -n istio-ingressgateway-2 apply -f -
kubectl -n istio-ingressgateway-1 create secret tls gateway-tls --cert=istio-ingressgateway-1.crt --key=istio-ingressgateway-1.key
kubectl -n istio-ingressgateway-2 create secret tls gateway-tls --cert=istio-ingressgateway-2.crt --key=istio-ingressgateway-2.key
kubectl -n istio-ingressgateway-1 apply -f gateway.yaml
kubectl -n istio-ingressgateway-2 apply -f gateway.yaml
```

If you execute `istioctl proxy-status` you can see that the ingress gateways are stale. Delete the pods to restart the gateways:
```
kubectl -n istio-ingressgateway-1 delete pod -l app=istio-ingressgateway
kubectl -n istio-ingressgateway-2 delete pod -l app=istio-ingressgateway
```

You can see that the pods don't become ready. `istioctl proxy-status` shows that the gateways aren't synced.

You'll find attached the [logs](https://github.com/istio/istio/files/4771798/logs.txt) of a gateway and its [config dump](https://github.com/istio/istio/files/4771797/config-dump.txt).

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```
$ istioctl version --remote
client version: 1.6.1
control plane version: 1.6.1
data plane version: 1.6.1 (3 proxies)
```
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.10-eks-bac369", GitCommit:"bac3690554985327ae4d13e42169e8b1c2f37226", GitTreeState:"clean", BuildDate:"2020-02-21T23:37:18Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**

Using istioctl with the attached operator manifest.

**Environment where bug was observed (cloud vendor, OS, etc)**

On AWS EKS, reproducible on Minikube.

**Comment by cober** [comment]
We've hit the same issue @yann-soubeyrand. A workaround for us that worked was to generate a manifest upfront, modify it (we also use another name instead of istio-ingressgateway) and remove istio selector in that manifest.

And on gateway side we mapped it to specific one by doing that:
```  labels:
    app: ab-ingressgateway
spec:
  selector:
    app: ab-ingressgateway```

**Comment by howardjohn** [comment]
Interesting, looks like we create two complete HTTP listeners which causes the conflict

**Comment by yann-soubeyrand** [comment]
This bug can be easily worked around, but it's a reliability issue that setting up a new ingress gateway in another namespace can break an existing working ingress gateway (that's what happened to me, hopefully on a test environment).

**Comment by howardjohn** [comment]
Not stale

**Comment by howardjohn** [comment]
Similar (probably same)

```yaml
apiVersion: networking.istio.io/v1beta1
kind: Gateway
metadata:
  name: ingressgateway
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - '*.example.com'
    port:
      name: https
      number: 443
      protocol: HTTPS
    tls:
      mode: PASSTHROUGH
---
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: vs1
  namespace: default
spec:
  gateways:
  - istio-system/ingressgateway
  hosts:
  - mysite.example.com
  tls:
  - match:
    - port: 443
      sniHosts:
      - mysite.example.com
    route:
    - destination:
        host: mysite.default.svc.cluster.local
        port:
          number: 443

---
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: vs2
  namespace: default
spec:
  gateways:
  - istio-system/ingressgateway
  hosts:
  - mysite.example.com
  http:
  - match:
    - uri:
        prefix: /
    route:
    - destination:
        host: mysite-svc
        port:
          number: 443
  tls:
  - match:
    - port: 443
      sniHosts:
      - mysite.example.com
    route:
    - destination:
        host: mysite.default.svc.cluster.local
        port:
          number: 443
```


We should:
1. Not fail with a NACK, and instead pick one of them
2. Add an analyzer for this scenario

**Comment by howardjohn** [comment]
In 1.9, this can be accomplished with the following IstioOperator: https://gist.github.com/howardjohn/aeea8da5f86d157d998ac2cf3cb48f59

Note we will plan to make this more stable and possibly automatic in the future

**Comment by brgsousa** [comment]
I have a similar issue, I tried following instructions described here https://istio.io/latest/docs/ops/common-problems/network-issues/#404-errors-occur-when-multiple-gateways-configured-with-same-tls-certificate but it didn't solve it. Each application is in a different namespace, if I set a single gateway (with a wildcard host *.test.com) the gateway can't "connect" to the virtualservice since they are in different namespaces.

**Comment by brgsousa** [comment]
> brgsousa commented 2 hours ago
I have a similar issue, I tried following instructions described here https://istio.io/latest/docs/ops/common-problems/network-issues/#404-errors-occur-when-multiple-gateways-configured-with-same-tls-certificate but it didn't solve it. Each application is in a different namespace, if I set a single gateway (with a wildcard host *.test.com) the gateway can't "connect" to the virtualservice since they are in different namespaces.

I was wrong. I could solve my problem simply by setting the gateway's namespace in my virtualservice (https://istio.io/latest/docs/reference/config/networking/virtual-service/#VirtualService)
